### ISSUE TITLE
Mixture Density Networks with ngboost

### ISSUE THREAD

**Issue by astrogilda** [stanfordmlgroup/ngboost#134]
Hey @alejandroschuler! Is it possible to create an MDN with ngboost? That is combine multiple distributions (in the most general case, from different families) to predict the output of interest?

**Comment by alejandroschuler** [comment]
Yes, in theory, but you would need to [create a new distribution class](https://stanfordmlgroup.github.io/ngboost/5-dev.html#Adding-Distributions) (e.g. ThreeNormalMixture, which would have three mean parameters and three variance parameters). It would be cool to have a class factory `mixture(Dist, K)` that generates the appropriate distribution class on-the-fly.  Honestly not sure off the top of my head if that's possible or easy.

**Comment by JoGall** [comment]
Did either of you ever experiment with implementing this @astrogilda or @alejandroschuler? And if not, did you ever find another framework capable of probabilistic forecasting with multimodal output distributions?
